os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -497,56 +487,37 @@ static inline void print_assert_detail(const uint8_t *filename, int lineno, stru
-	} else {
-		lldbg_noarg("##########################################################################################################################################\n");
-		lldbg_noarg("\n\n");
-		/* treat kernel fault */
-		arm_assert();
-	}
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("\n");
+	up_dumpstate(fault_tcb, asserted_location);
 	
 	/* Dump the state of all tasks (if available) */
 	task_show_alivetask_list();
-	lldbg_noarg("\n");
-
-	/* Dump the asserted TCB */
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("Asserted TCB Info\n");
-	lldbg_noarg("=====================================================================\n");
-	task_show_tcbinfo(fault_tcb);
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("\n");
 
-#ifdef CONFIG_APP_BINARY_SEPARATION
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("Application Info\n");
-	lldbg_noarg("=====================================================================\n");
-	elf_show_all_bin_section_addr();
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("\n");
-#endif
 
-	up_dumpstate(fault_tcb, asserted_location);
 
 #ifdef CONFIG_APP_BINARY_SEPARATION
+	elf_show_all_bin_section_addr();
 	if (IS_FAULT_IN_USER_THREAD(fault_tcb)) {
-		lldbg_noarg("=====================================================================\n");
-		lldbg_noarg("Checking current app heap for corruption...\n");
-		lldbg_noarg("=====================================================================\n");
+		lldbg_noarg("===========================================================App_Heap_Corruption==========================================\n");
+		lldbg("Checking current app heap for corruption...\n");
 		if (mm_check_heap_corruption((struct mm_heap_s *)(fault_tcb->uheap)) == OK) {
-			lldbg_noarg("No app heap corruption detected\n");
+			lldbg("No app heap corruption detected\n");
 		}
-		lldbg_noarg("=====================================================================\n");
 	}
 #endif
 
+	/* Dump the asserted TCB */
+	task_show_tcbinfo(fault_tcb);
+
+	lldbg_noarg("===========================================================Kernel_Heap_Corruption==========================================\n");
+	lldbg("Checking kernel heap for corruption...\n");
+	if (mm_check_heap_corruption(g_kmmheap) == OK) {
+		lldbg("No kernel heap corruption detected\n");
+	} else {
+		lldbg_noarg("##########################################################################################################################################\n");
+		/* treat kernel fault */
+		arm_assert();
+	}
+
 #if defined(CONFIG_BOARD_CRASHDUMP)
 	lldbg_noarg("Perform Crashdump\n");
 	board_crashdump(up_getsp(), fault_tcb, (uint8_t *)filename, lineno);
```

Comment:
Could you please confirm two additional things?

1. Please set the log output order.
- Currently, the order is not organized and jumbled (e.g., user heap, kernel heap).
- Rearrange the order in a meaningful way so that the logs displayed are easy for us to analyze.

2. There should be only informational log output in print_assert_detail().
- Move non-informational log outputs out of the function (e.g., board_autoled_on(), reboot_reason_write_user_intended(), board_crashdump()).

Reply:
1.  We have set the log output order as specified in pr desc.

- We have made it pretty much more organized and clean.
- We have made the log order and format same for both v8-m and v7-a.

2. Now, print_assert_detail() function will only have informational log output.